draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -346,153 +415,266 @@ authorized(AS x, AS y) =  / Else, "Provider+" if the VAP-SPAS entry
+        For an upstream or downstream path, if the ASPA-based AS_PATH verification did not determine the route status to be Invalid or Valid (<xref target="invalid"/>, <xref target="valid"/>, respectively), then it is Unknown.
       </t>
     </section>
-
+    </section>
     <section title="Algorithm for Upstream Paths" anchor="Upflow">
       <t>
-        The upstream verification algorithm described here is applied when a route is received from a customer or lateral peer, or is received by an RS from an RS-client, or is received by an RS-client from an RS.
-        In all these cases, the receiving/validating eBGP router expects the AS_PATH does not have a down-ramp, only up-ramp.
-        Thus max_down_ramp and min_up_ramp are set to 0.
+        The upstream verification algorithm specified here is applied when a route is received from a customer or lateral peer, or received by an RS from an RS-client, or received by an RS-client from an RS.
+        <!-- In all these cases, the receiving/validating router expects the AS_PATH to consist of only an up-ramp, i.e., C2P hops successively from the origin AS to the neighbor AS (most recently added). -->
       </t>
       <t>
-        The upstream path verification procedure is specified as follows:
+        Following the principles described in <xref target="invalid"/>, consider the AS_PATH in the received route represented by {AS(N), AS(N-1),..., AS(2), AS(1)} and find the minimum I (1 &lt;= I &lt;= N-1) for which hop(AS(I), AS(I+1)) = "Not Provider+".
+        If such minimum I does not exist, then set it equal to N (i.e., AS_PATH length).
+        Set NP_right equal to the minimum I value.  
+        Now following the principles described in <xref target="valid"/>, compute the minimum value of m (1 &lt;= m &lt;= N-1) for which hop(AS(m), AS(m+1)) = "No Attestation" or "Not Provider+".
+        If such minimum m does not exist, then set it equal to N
+        Set upramp_length equal to the minimum m value.
+        <!--
+        The route is Invalid if NP_right &lt; N; else, it is not Invalid.
+        If the route was determined not Invalid, then the route is Valid if upramp_length = N; else it is Unknown.
+        -->
+      </t>
+      <t> 
+        The step-wise upstream path verification procedure is specified as follows:
       </t>
       <t>
         <list style="numbers">
           <t>
             If the AS_PATH has an AS_SET, then the procedure halts with the outcome "Invalid".
+            Else, continue.
+          </t>
+          <t>
+            Let the AS_PATH be represented by {AS(N), AS(N-1), ..., AS(2), AS(1)} in terms of unique ASNs.
           </t>
           <t>
             If N = 1, then the procedure halts with the outcome "Valid".
             Else, continue.
           </t>
           <t>
-            At this step, N &ge; 2. If max_up_ramp is less than N the procedure halts with the outcome 'Invalid'.
+            At this step, N &ge; 2. 
```

Comment:
You are again ignoring my comments from the previous review. max and min have exact semntics. It is important to have variables prooperly named.

Reply:
@mitradir 
I have said this before. I think we should first describe route leak and the principle of how "Invalid (route leak) is found using ASPA (see new Sec. 7.1.1). (Rather than jump too soon to the inverted V picture and the ramps.) 
 
   "... for the AS path to be Invalid (route leak), there must be two hops -- one on the
   left and another on the right -- facing each other and each one is not C2P
   (i.e., each is P2C or P2P)."  (new Section 7.1.1)

In finding this condition using ASPA, NP_left and NP_right is more suitable/intuitive terminology (which map to your max_down_ramp and max_up_ramp).  NP = Not Provider. We can do a zoom call to discuss and clarify this and more about Sec. 7.

Reply:
@mitradir @cjeker 
I will make a Commit of the changes tomorrow.